projects/core/src/auth/user-auth/events/user-auth-event.builder.spec.ts

```diff
@@ -1,52 +1,90 @@
 import { TestBed } from '@angular/core/testing';
 import { Action, ActionsSubject } from '@ngrx/store';
-import { EventService } from 'projects/core/src/event';
-import { Subject } from 'rxjs';
+import { BehaviorSubject, Subject } from 'rxjs';
 import { take } from 'rxjs/operators';
+import { EventService } from '../../../event/event.service';
+import { AuthService } from '../facade/auth.service';
 import { AuthActions } from '../store/actions';
 import { UserAuthEventBuilder } from './user-auth-event.builder';
 import { LoginEvent, LogoutEvent } from './user-auth.events';
 
+const isUserLoggedIn$ = new BehaviorSubject<boolean>(false);
+class MockAuthService implements Partial<AuthService> {
+  isUserLoggedIn = () => isUserLoggedIn$.asObservable();
+}
+
 describe('UserAuthEventBuilder', () => {
-  let eventService: EventService;
   let actions$: Subject<Action>;
+  let eventService: EventService;
 
   beforeEach(() => {
     actions$ = new Subject();
     TestBed.configureTestingModule({
-      providers: [{ provide: ActionsSubject, useValue: actions$ }],
+      providers: [
+        { provide: ActionsSubject, useValue: actions$ },
+        { provide: AuthService, useClass: MockAuthService },
+      ],
     });
 
     TestBed.inject(UserAuthEventBuilder); // register events
     eventService = TestBed.inject(EventService);
   });
 
   describe('LogoutEvent', () => {
```

Comment:
Nice set of tests.
It would be good if we could explicitly test the ASM agent login / logout flow, and assert that it doesn't trigger the user login / logout events.